### PR TITLE
Mesh init 5/6 attributes

### PR DESCRIPTION
**Pull request by Rahul-JOON**
This PR sets up the base for the mesh class in pyelastica as per issue #270 

This PR contains basic code for reading an stl file and testing for the faces array's geometry that is generated.

### CODE REVIEW

File: mesh/mesh_initializer.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    How to initialize a mesh?
+    -------------------------
+
+    mesh = MeshInitialize(r('<filepath>'))
+
+    PS: Please be sure to add .stl / .obj at the end of the filepath, if already present, ignore.
+    """
+    def __init__(self, filepath):
+        self.mesh = pv.read(filepath)
+        self.pyvista_faces = self.mesh.faces
```
Comment (armantekinalp) [inline mesh/mesh_initializer.py:21]: why do you name as pyvista_faces and not faces?

File: mesh/mesh_initializer.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        self.number_of_faces = self.mesh.n_faces
+        self.pyvista_points = self.mesh.points
```
Comment (armantekinalp) [inline mesh/mesh_initializer.py:23]: Same as above

File: assets/cube.stl
Comment (armantekinalp) [inline assets/cube.stl:1]: I will get back to you where to keep stl file

**Review by armantekinalp — CHANGES_REQUESTED** [review]
Some minor questions/comments.

File: mesh/mesh_initializer.py (hunk 1 above)
Comment (Rahul-JOON) [reply mesh/mesh_initializer.py:21]: I named it pyvista_faces/points to specify that it's the faces array from pyvista library, I later used this array and established a self.faces array that stores the data in the desired format and dimension as requested by Ali in #270.

File: assets/cube.stl
Comment (Rahul-JOON) [reply assets/cube.stl:1]: Okay, Thanks!

File: mesh/mesh_initializer.py (hunk 1 above)
Comment (Rahul-JOON) [reply mesh/mesh_initializer.py:21]: I have added a detailed explanation of the function and class works as comments in the next commit; I will push that soon after this PR is resolved/merged.

File: assets/cube.stl
Comment (armantekinalp) [reply assets/cube.stl:1]: @Rahul-JOON please move this file inside the tests.

Comment (armantekinalp) [reply assets/cube.stl:1]: Also is stl files always binary? Shouldn't it be readable

Comment (Rahul-JOON) [reply assets/cube.stl:1]: Sure, I will move the file.
No, the stl file comes in both a human-readable ASCII format or a smaller binary file format; to reduce the file size, I chose the binary one.

**Review by Ali-7800 — APPROVED** [review]
LGTM

File: mesh/mesh_initializer.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    """
+    This Mesh Initializer class uses pyvista to import mesh files in the
+    STL or OBJ file formats and initializes the necessary mesh information.
+
+    How to initialize a mesh?
+    -------------------------
+
+    mesh = MeshInitialize(r('<filepath>'))
+
+    PS: Please be sure to add .stl / .obj at the end of the filepath, if already present, ignore.
```
Comment (bhosale2) [inline mesh/mesh_initializer.py:17]: Add this under Notes and not PS

File: mesh/mesh_initializer.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
+__doc__ = """ Mesh Initializer using Pyvista """
+
+import pyvista as pv
+import numpy as np
+
+
+class MeshInitialize():
```
Comment (bhosale2) [inline mesh/mesh_initializer.py:7]: why call it `MeshInitialize` and not just `Mesh`?

File: mesh/mesh_initializer.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        self.mesh = pv.read(filepath)
+        self.pyvista_faces = self.mesh.faces
+        self.number_of_faces = self.mesh.n_faces
+        self.pyvista_points = self.mesh.points
+        self.faces = self.face_calculation(self.pyvista_faces, self.pyvista_points, self.number_of_faces)
+
+    def face_calculation(self, pvfaces, meshpoints, n_faces):
+        """
+        This function converts the faces from pyvista to pyelastica geometry
+        """
```
Comment (bhosale2) [inline mesh/mesh_initializer.py:29]: @armantekinalp  I was thinking that here on wards we should try to make it a habit of adding type hints and including the information in doc string as needed. Ofc eventually we need to propagate this backward in `pyelastica` as well. Thoughts?

**Review by bhosale2 — CHANGES_REQUESTED** [review]
Minor comments

File: mesh/mesh_initializer.py (hunk 3 above)
Comment (Rahul-JOON) [reply mesh/mesh_initializer.py:17]: Okay, I will add this change in the next PR.

File: mesh/mesh_initializer.py (hunk 4 above)
Comment (Rahul-JOON) [reply mesh/mesh_initializer.py:7]: The names are truly subjective, I can change it to Mesh if that's better.

File: mesh/mesh_initializer.py (hunk 5 above)
Comment (armantekinalp) [reply mesh/mesh_initializer.py:29]: I agree with you @bhosale2 . We will take care already published 'pyelastica' code base later but new code can have type hints.  @Rahul-JOON can you add type hints for the new code in this PR.

Comment (Rahul-JOON) [reply mesh/mesh_initializer.py:29]: Yes, sure I will add them.

File: mesh/mesh_initializer.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+class Mesh():
+    """
+    This Mesh Initializer class uses pyvista to import mesh files in the
+    STL or OBJ file formats and initializes the necessary mesh information.
+
+    How to initialize a mesh?
+    -------------------------
+
+    mesh = MeshInitialize(r"<filepath>")
```
Comment (bhosale2) [inline mesh/mesh_initializer.py:15]: correct the docs as well, to be consistent with class name change

File: mesh/mesh_initializer.py [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+    mesh.mesh_scale:
+        - Stores the 3 dimensions of the smallest box that could fit the mesh.
+        - Dimension: (3 spatial lengths)
+
+    mesh.mesh_center:
+        - Stores the coordinates of the position vector of the center of the smallest box that could fit the mesh.
+        - Dimension: (3 spatial coordinates)
+    """
+    def __init__(self, filepath: str):
```
Comment (bhosale2) [inline mesh/mesh_initializer.py:45]:
```suggestion
    def __init__(self, filepath: str) -> None:
```

**Review by Ali-7800 — APPROVED** [review]
LGTM

**Review by bhosale2 — APPROVED** [review]
LGTM

**Pull request merged**